broker/src/main/java/io/zeebe/broker/engine/impl/SubscriptionApiCommandMessageHandlerService.java

```diff
@@ -77,4 +82,31 @@ protected void onActorStarting() {
                     }));
     return future;
   }
+
+  @Override
+  public void onDiskSpaceNotAvailable() {
+    actor.call(
+        () -> {
+          LOG.debug(
+              "Broker is out of disk space. All requests with topic {} will be rejected.",
+              SUBSCRIPTION_TOPIC);
+          atomix.getCommunicationService().unsubscribe(SUBSCRIPTION_TOPIC);
+          atomix
```

Comment:
Here it doesn't make a difference because the sender is not expecting any response. Ideally this should respond with a "Resource exhausted" error similar to the deployment request. But the current interface for subscription message request returns nothing. This would probably be fixed by #4786